pkg/appliance/checks.go

```diff
@@ -172,10 +177,26 @@ func CheckVersions(ctx context.Context, stats openapi.StatsAppliancesList, appli
 				statV, err := ParseVersionString(stat.GetVersion())
 				if err != nil {
 					log.Warn("failed to parse version from stats")
+					skip = append(skip, SkipStruct{
+						Appliance: appliance,
+						Reason:    "failed to parse version from stats",
+					})
+					continue
+				}
+				res, err := CompareVersionsAndBuildNumber(statV, v)
+				if err != nil {
+					log.Warn("failed to compare versions")
+					skip = append(skip, SkipStruct{
```

Comment:
no need to log and return error.  remove log line here and :179-180 .

If you really want to log the value, log once in caller instead of duplicate logs here.

Reply:
We do not return the error here though, so not duplicate at all. The error just means we skip this appliance and the only case where we would return these errors is if all the appliances are skipped for some reason. If that does not happen, this is the only place where we actually log the reason, otherwise it's only printed in the summary.